bagua-core-internal/src/communicators/mod.rs

```diff
@@ -906,7 +906,7 @@ impl BaguaCommunicatorInner {
         );
         let communicator_ptr = self.comm_ptr;
         let tensor_ptr = tensor.data_ptr();
-        let count = tensor.num_elements_allocated();
```

Comment:
we should do safety check here to ensure it is divisible?

Reply:
refer to line 902